client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx

```diff
@@ -217,13 +280,68 @@ const PlanOnlyThankYou = ( {
 					/>
 				}
 				footerDetails={ footerDetails }
+				upsellProps={ upsellProps }
 			/>
 		</>
 	);
 };
 
-export default connect( null, {
-	errorNotice,
-	removeNotice,
-	successNotice,
-} )( PlanOnlyThankYou );
+function getBusinessPlanSlugToUpgrade( currentPlanSlug: string ) {
+	const currentPlan = getPlan( currentPlanSlug );
+	const planKey = findFirstSimilarPlanKey( PLAN_BUSINESS, { term: currentPlan?.term } );
+
+	// It's just for TS, since .find() can return undefined
+	if ( ! planKey ) {
+		return;
+	}
+
+	return getPlan( planKey );
+}
+
+export default connect(
+	( state: IAppState, props: PlanOnlyProps ) => {
+		let promotePlanData;
+		if ( isPersonal( props.primaryPurchase ) || isPremium( props.primaryPurchase ) ) {
+			const promotePlan = getBusinessPlanSlugToUpgrade( props.primaryPurchase.productSlug );
+			const promotePlanSlug = promotePlan?.getStoreSlug() || '';
+
+			const pathSlug = promotePlan?.getPathSlug?.();
+			const selectedSiteId = getSelectedSiteId( state ) || 0;
+			const currencyCode = getCurrentUserCurrencyCode( state );
+
+			const businessPrice = getSitePlanRawPrice( state, selectedSiteId, promotePlanSlug, {
+				returnMonthly: false,
+			} );
+			const businessDiscountPrice = getPlanDiscountedRawPrice(
+				state,
+				selectedSiteId,
+				promotePlanSlug,
+				{
+					returnMonthly: false,
+				}
+			);
+
+			// The condition is just for TS fix
+			if ( businessPrice && businessDiscountPrice && currencyCode && promotePlan && pathSlug ) {
```

Comment:
That's a long condition, I wonder if we shouldn't break it down - something that in addition to improving readability will also help a bit with performance (although marginally):

```javascript
const promotePlan = getBusinessPlanSlugToUpgrade( props.primaryPurchase.productSlug );

if ( ! promotePlan ) {
  return {};
}

const currencyCode = getCurrentUserCurrencyCode( state );

if ( ! currencyCode ) {
  return {};
}
```

Reply:
The main point is - if we have `undefined` somewhere - it's a bug, it should be `undefined` :) We should always have information about prices, currency and business plan slug

Reply:
I wouldn't consider that as a bug - checking for `undefined` values is also a way to handle transient timing issues with network requests (e.g. if the API request that retrieves the list of plans takes longer than usual). That's why I find this comment confusing: it infers that this condition was added just because of some TypeScript issue whereas we should really check that those values are indeed available.